MidiKeyboard/src/com/mobileer/midikeyboard/MainActivity.java

```diff
@@ -39,7 +44,24 @@
 
     private MidiInputPortSelector mKeyboardReceiverSelector;
     private MusicKeyboardView mKeyboard;
+    private Button mProgramButton;
     private MidiManager mMidiManager;
+    private int mChannel; // ranges from 0 to 15
+    private int[] mPrograms = new int[MidiConstants.MAX_CHANNELS]; // ranges from 0 to 127
+    private byte[] mByteBuffer = new byte[3];
+
+    public class ChannelSpinnerActivity extends Activity implements AdapterView.OnItemSelectedListener {
```

Comment:
Usually, I don't bother creating a named class for a Listener, I just instantiate a new object in the .setOnItemSelectedListener()